cmd/xl-sets.go

```diff
@@ -214,16 +226,26 @@ func (s *xlSets) connectDisksAndLockers() {
 		}
 		disk.SetDiskID(format.XL.This)
 		s.xlDisksMu.Lock()
-		s.xlDisks[i][j] = disk
-		s.xlLockers[i][j] = s.lockersMap[endpoint]
+		s.xlDisks[setIndex][diskIndex] = disk
+		s.xlLockers[setIndex][diskIndex] = s.lockersMap[endpoint]
 		s.xlDisksMu.Unlock()
+
+		// Send a new disk connect event with a timeout
+		select {
+		case s.disksConnectEvent <- diskConnectInfo{setIndex: setIndex}:
+		case <-time.After(100 * time.Millisecond):
```

Comment:
@vadmeste This will timeout only when there is no event received in `s.disksConnectEvent <- diskConnectInfo{setIndex: setIndex}` . Is that what we needed here? If not, maybe the comment is misleading

Reply:
```
package main

import "fmt"
import "time"

func callselect(c chan int) {
	for {
		select {
		case c <- 1:
		case <-time.After(100 * time.Millisecond):
			fmt.Println("timed")
		}
	}
}

func main() {
	c := make(chan int)
	i := 0
	go func() {
		for (i < 20) {
			fmt.Println(<-c)
			i++
		} 
	}()
	callselect(c)
}
```

Reply:
> This will timeout only when there is no event received in `s.disksConnectEvent <- diskConnectInfo{setIndex: setIndex}` . Is that what we needed here? If not, maybe the comment is misleading

@Praveenrajmani I addessed here Harsha concern that we always should send a message in a channel with a timeout. 

So the goal of the select block here is to try to send a new disk connect event during 100 ms, otherwise just quit